### PR TITLE
add submission form, database, and validator for image URL

### LINKED ISSUE #2: Save Image Link
#### As a user I want to save a link to an image.

__Story__:

Now that we have an awesome landing page for our users to experience it is time
to introduce our first feature! In general, we want to deliver value to our
customers as fast as possible. As a result, we have decided that we want our
users to be able to save links to images, and immediately see that we have
saved the URL to the image.

One could argue that simply saving the image, and not showing it, provides some
value. Yes, you could split this story here. However, because saving image URLs
and viewing those images are both relatively small additions that work well
together, it is acceptable to complete them together. Note, on the other hand,
that we do not yet want to implement the image index. Its addition adds too
much complexity, so we’ll save that for a future story.

In general, feature addition, or stories, should be split into the smallest
deliverable chunk that provides value to your users with the caveat that it
might make sense to combine two, sometimes more, tiny and related features into
a single story.

__Acceptance criteria__:
- [ ] The landing page links to the image link submission form.
- [ ] The link is entered through a form.
- [ ] After the form is saved, the link is persisted in the database.
- [ ] After the form is submitted, the user is redirected to a page displaying
  the image.
- [ ] I cannot successfully save an image with an invalid URL.
- [ ] An error message is associated with the appropriate input field on
  failure.

__Discussion Topic__:
- [ ] What is a valid image URL?

__Dependencies__:
- Deployment Environment

__Note__: Many image sharing sites will permit one to upload images, or
download images when provided a link. For this project we will only work with
the link, i.e., URL, to various images.

__Note__: To understand the database tables created by the `acts_as_taggable` gem, you may find it useful to use a program like the [DB Browser for SQLite](http://sqlitebrowser.org/).

### PR DESCRIPTION
**Pull request by Brick-Mover**
Closes #2 

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [ ] I have replaced `X` above with the associated issue number.
- [ ] I have changed the issue's label to "pull request".
- [ ] I have assigned the designated reviewer to this PR.
- [ ] I have set the appropriate label on this PR.
- [ ] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

**Comment by Brick-Mover** [comment]
https://trello.com/c/a6BltghR/27-issue-2-save-image-link

File: db/schema.rb [hunk 1]
```diff
@@ -10,6 +10,18 @@
...
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 0) do
+ActiveRecord::Schema.define(version: 2019_02_06_224303) do
+
+  create_table "image_urls", force: :cascade do |t|
+    t.string "link"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
```
Comment (blakeherrington) [inline db/schema.rb]: It looks like maybe this is out of date? Did you try having a `image_urls` table and then change the name later?

File: test/models/image_test.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+class ImageTest < ActiveSupport::TestCase
+  test 'non-image URL' do
+    image = Image.new(link: 'www.foo.pdf')
+    assert_not image.valid?
+  end
+
+  test 'image URL' do
+    image = Image.new(link: 'www.foo.jpeg')
+    assert image.valid?
+  end
```
Comment (blakeherrington) [inline test/models/image_test.rb]: Are these two tests sufficient to test your regex?

File: test/models/image_test.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+require 'test_helper'
+
+class ImageTest < ActiveSupport::TestCase
+  test 'non-image URL' do
+    image = Image.new(link: 'www.foo.pdf')
+    assert_not image.valid?
```
Comment (blakeherrington) [inline test/models/image_test.rb]: When it is not valid, should we also be testing that the error message is set properly?

File: app/controllers/images_controller.rb [hunk 4]
```diff
@@ -1,3 +1,26 @@
...
   def index; end
+
+  def new
+    @image = Image.new
+  end
+
+  def create
+    @image = Image.new(image_params)
+    if @image.save
+      redirect_to @image, notice: 'Image url was successfully created.'
```
Comment (blakeherrington) [inline app/controllers/images_controller.rb]: What does this `notice` do?

File: app/views/images/index.html.erb [hunk 5]
```diff
@@ -1,2 +1,4 @@
 <h1>Hello</h1>
 <p>World</p>
+
+<%= link_to "Submit image url", new_image_url, method: :get %>
```
Comment (blakeherrington) [inline app/views/images/index.html.erb]: Minor - this `method` option is not required here. I believe the default method is `get`

**Pull request merged**